OpenEdXMobile/src/main/java/org/edx/mobile/base/FindCoursesBaseActivity.java

```diff
@@ -87,6 +105,62 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    /**
+     * Simply loads a url within a WebView for Marshmallow & above and differently in case of
+     * Lollipop & below.<br/>
+     * WebViews prior to Marshmallow (API Level 23) don't provide a way to get the HTTP status
+     * codes if the url being loaded fails.<br/>
+     * This utility function solves this by making a server call using the url provided to query if
+     * an error is being return and show error or move on to load the url in WebView, if the server
+     * is responding correctly.
+     *
+     * @param url The URL to load.
+     */
+    protected void loadUrlBasedOnOsVersion(@NonNull final String url) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            errorNotification.hideError();
+            showLoadingProgress();
+            okHttpClientProvider.get().newCall(new Request.Builder()
```

Comment:
**nit:** Shouldn't we refactor this server call to a task to make it reusable as it will require on other places too.